File: 프로그래머스/2/42586. 기능개발/기능개발.go
import "math"

func solution(progresses []int, speeds []int) []int {
    var list, succ, ans []int
    for i:=0;i<len(progresses);i++{
        list = append(list, int(math.Ceil(float64(100-progresses[i])/float64(speeds[i]))))
    }
    succ = append(succ, list[0])
    list = list[1:]
    for i:=0; i<len(list);i++{
        if succ[0] >= list[i]{
            succ = append(succ,list[i])
        }else{
            ans = append(ans, len(succ))
            succ = nil
            succ = append(succ, list[i])
        }
    }
    ans = append(ans, len(succ))
    return ans
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    complete = []
    for p, s in zip(progresses, speeds):
        complete.append(math.ceil((100-p)/s))
    que = [complete.pop(0)]
    for c in complete:
        if que[0] >= c:
            que.append(c)
        else:
            answer.append(len(que))
            que.clear()
            que.append(c)
    answer.append(len(que))
    return answer